### PR TITLE
remove shebang when turning ecmascript modules into chunk items

### PR DESCRIPTION
**Pull request by ForsakenHarmony**

### CODE REVIEW

**Comment by ForsakenHarmony** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3827** <a href="https://app.graphite.dev/github/pr/vercel/turbo/3827" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/vercel/turbo/3827?utm_source=stack-comment).

File: crates/turbopack-ecmascript/src/transform/mod.rs [hunk 1]
```diff
@@ -261,6 +261,17 @@ impl EcmascriptInputTransform {
     }
 }
 
+pub fn remove_shebang(program: &mut Program) {
```
Comment (jridgewell) [inline crates/turbopack-ecmascript/src/transform/mod.rs:264]: Can we add a test case to `turbopack-tests`?

**Review by jridgewell — APPROVED** [review]

**Review by sokra — APPROVED** [review]

File: crates/turbopack-ecmascript/src/lib.rs [hunk 2]
```diff
@@ -337,6 +338,8 @@ impl EcmascriptChunkItem for ModuleChunkItem {
                 }
                 program.visit_mut_with(&mut swc_core::ecma::transforms::base::hygiene::hygiene());
                 program.visit_mut_with(&mut swc_core::ecma::transforms::base::fixer::fixer(None));
+
+                remove_shebang(&mut program);
```
Comment (alexkirsz) [inline crates/turbopack-ecmascript/src/lib.rs:342]: Please add a comment explaining _why_ we're removing the shebang, either here or in the function itself.

**Review by alexkirsz — APPROVED** [review]

File: crates/turbopack-ecmascript/src/transform/mod.rs (hunk 1 above)
Comment (ForsakenHarmony) [reply crates/turbopack-ecmascript/src/transform/mod.rs:264]: done

File: crates/turbopack-ecmascript/src/lib.rs (hunk 2 above)
Comment (ForsakenHarmony) [reply crates/turbopack-ecmascript/src/lib.rs:344]: done

**Pull request merged**